apis/kueue/v1alpha1/cohort_types.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	kueuebeta "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+)
+
+// CohortSpec defines the desired state of Cohort
+type CohortSpec struct {
+	// Parent references the name of the Cohort's parent, if
+	// any. It satisfies one of three cases:
+	// 1) Unset. This Cohort is the root of its Cohort tree.
+	// 2) References a non-existent Cohort. We use default Cohort (no borrowing/lending limits).
+	// 3) References an existent Cohort.
+	//
+	//+kubebuilder:validation:MaxLength=253
+	//+kubebuilder:validation:Pattern="^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$"
+	//
+	Parent string `json:"parent,omitempty"`
+
+	// ResourceGroups describes groupings of Resources and
+	// Flavors.  Each ResourceGroup defines a list of Resources
+	// and a list of Flavors which provide quotas for these
+	// Resources. Each Resource and each Flavor may only form part
+	// of one ResourceGroup.  There may be up to 16 ResourceGroups
+	// within a Cohort.
+	//
+	// BorrowingLimit limits how much members of this Cohort
+	// subtree can borrow from the parent subtree. This limit must
+	// only be set when the Cohort has a parent.
+	//
+	// LendingLimit limits how much members of this Cohort
+	// subtree can lend to the parent subtree. This limit must
+	// only be set when the Cohort has a parent.
+	//
+	//+listType=atomic
+	//+kubebuilder:validation:MaxItems=16
+	//
+	ResourceGroups []kueuebeta.ResourceGroup `json:"resourceGroups,omitempty"`
+}
+
+// CohortStatus defines the observed state of Cohort
+type CohortStatus struct {
```

Comment:
Do you expect it to be left empty? If yes - drop the struct for now. If not - please add the expected content.

Reply:
Added the `Conditions` field, and the CohortActive field. One note: I modified the condition from the KEP slightly, to match [ClusterQueue](https://github.com/kubernetes-sigs/kueue/blob/cb92b429a5a711699fb3993cf8d095fd667e2d1b/apis/kueue/v1beta1/clusterqueue_types.go#L356-L358)

KEP
```
CohortActive = "CohortActive"
```

This PR
```
CohortActive = "Active"
```